include/boost/gil/detail/math.hpp

```diff
@@ -28,6 +48,158 @@ inline detail::kernel_2d<T, Allocator> get_identity_kernel()
+                        col_boundary >= 0 && col_boundary < convolved_kernel_size)
+                    {
+                        aux_total +=
+                            kernel2[flip_kernel_row][flip_kernel_col] *
+                            dummy_kernel[row_boundary][col_boundary];
+                    }
+                }
+            }
+            convolved_kernel[dummy_row][dummy_col] = aux_total;
+        }
+    }
+    return convolved_kernel;
+}
+
+/// \brief Fills kernel vector given as argument with a second order kernel in horizontal or
+/// vertical direction. The type of the kernel which is to be used for filling will be indicated 
+/// by the variable 'type'.
+/// \param kernel - Kernel vector which will be filled.
+/// \param type - Indicates the type of second order derivative kernel which is to be filled inside
+/// first argument.
+static void kernel_vector_fill(std::vector<std::vector<float>>& kernel, kernel_type type)
+{
+    if(type == kernel_type::sobel_dx)
+    {
+        for(std::ptrdiff_t row = 0;row < 5; ++row)
+        {
+            for(std::ptrdiff_t col = 0;col < 5; ++col)
+            {
+                kernel[row][col] = dx_sobel2[5 * row + col];
+            }
+        }
+    }
+    else if(type == kernel_type::sobel_dy)
+    {
+        for(std::ptrdiff_t row =0;row < 5; ++row)
+        {
+            for(std::ptrdiff_t col = 0;col < 5; ++col)
+            {
+                kernel[row][col] = dy_sobel2[5 * row + col];
+            }
+        }
+    }
+}
+
+/// \brief Passes parameters to 'kernel_convolve_impl()' repeatedly until kernel vector of desired
+/// order is obtained.
+/// \param order - Indicates order of derivative whose kernel vector is to be returned.
+/// \param type - Indicates the type of kernel vector which is to be returned.
+static auto kernel_convolve(unsigned int order, kernel_type type) -> std::vector<float>
+{
+    std::vector<float> convolved_kernel_flatten;
+    std::vector<std::vector<float>> convolved_kernel(5, std::vector<float>(5));
+    
+    kernel_vector_fill(convolved_kernel, type);
+
+    std::vector<std::vector<float>>smoothing_dummy = smoothing_kernel;
+
+    // Variable 'smooth_repetition' will store the number of times we need to convolve 
+    // 'smoothing_dummy' with itself. This number when used as a power of 2 in its exponentiation,
+    // will result in a number which is the largest power of 2 smaller than 'order - 2'.
+    unsigned int smooth_repetition = (unsigned int)log2(order - 2);
+
+    for(unsigned int i = 0;i < smooth_repetition; ++i)
+    {
+        smoothing_dummy = kernel_convolve_impl(smoothing_dummy, smoothing_dummy);
+    }
+
+    convolved_kernel = kernel_convolve_impl(convolved_kernel, smoothing_dummy);
+
+    // Variable 'order_decrease' will store the amount of decrease in order obtained due to the above 
+    // optimization. It stores the largest power of 2 smaller than 'order - 2'.
+    unsigned int order_decrease = pow(2, smooth_repetition);
```

Comment:
missing `std::` for `pow`

Reply:
I have implemented this change. Just for educational purposes, are we making this change to make the code more adept into c++11 or is it somehow related to performance?

Reply:
Well, this code is in public headers, so we need to keep it clean ~C++11~ C+14